src/getEdgeMassMatrix.jl

```diff
@@ -275,11 +276,51 @@ function merge(a, b)
 c      = copy(a)
 idz    = c .== 0
 c[idz] = b[idz]
-
 return c
+end
+
+function DerivativeTimesVector(M::OcTreeMeshFV, v::Vector,
+                               x::Vector)
+   # Derivative (getdEdgeMassMatrix) times a vector(x)
+   if isempty(M.Pe)
+      M.Pe = getEdgeMassMatrixIntegrationMatrix(M.S,M.h)
+   end
+
+  # dM = M.Pe'* sdiag(M.Pe*v) * kron(ones(24,1),speye(nnz(M.S)))
+
+   pv = M.Pe * v
 
+   dMw = M.Pe' * (pv .* repmat(x,24))
+   
+   return dMw
+end  # function DerivativeTimesVector
+
+function DerivativeTrTimesVector(M::OcTreeMeshFV, v::Vector,
+                                 x::Vector)
+   # Derivative (getdEdgeMassMatrix) transpose times a vector(x)
+   if isempty(M.Pe)
+      M.Pe = getEdgeMassMatrixIntegrationMatrix(M.S,M.h)
 end
 
+  # dM' = kron(ones(24,1),speye(nnz(M.S)))' * sdiag(M.Pe*v) * M.Pe
+
+   pv = M.Pe * v
+   dd = pv .* (M.Pe * conj(x)) 
+   pv = [] 
+
+   ns = nnz(M.S)
+   dMTw = dd[1:ns]
+
+   j = ns
+   for i = 2:24
+      @inbounds dMTw += dd[j+1 : j+ns]
+      j += ns
+   end  # i
+   dMTw = conj(dMTw)
+
+   return dMTw
+end  # function DerivativeTrTimesVector
+
 function getNodesFromIndices(sv,mm,i0::Vector{Int},j0::Vector{Int},k0::Vector{Int})
```

Comment:
`getNodesFromIndices` is defined in `getEdgeMassMatrix.jl` and `getFaceMassMatrix.jl`, which generates the following warning:
```
julia> using JOcTree
WARNING: Method definition getNodesFromIndices(Any, Any, Array{Int64, 1}, Array{Int64, 1}, Array{Int64, 1}) in 
module JOcTree at /home/patrick/.julia/v0.5/JOcTree/src/getEdgeMassMatrix.jl:326 overwritten at 
/home/patrick/.julia/v0.5/JOcTree/src/getFaceMassMatrix.jl:127.
```
Perhaps we could rename the file `diagMUtils.jl` to `JOcTreeUtils.jl`and throw `getNodesFromIndices` (and other little helper functions you might come across) in there?